icu4c/source/test/intltest/dtfmttst.cpp

```diff
@@ -5540,6 +5541,17 @@ void DateFormatTest::TestParseRegression13744() {
     assertEquals("Error index", inDate.length(), pos.getErrorIndex());
 }
 
+void DateFormatTest::TestAdoptCalendarLeak() {
```

Comment:
Thank for adding a test Frank.
I wonder if it might be worth adding some comments like:
This test relies on the locale fullName exceeding ULOC_FULLNAME_CAPACITY in order for setKeywordValue to fail.

ULOC_FULLNAME_CAPACITY